deepmd/dpmodel/output_def.py

```diff
@@ -275,19 +315,23 @@ def do_derivative(
     def_derv_c: Dict[str, OutputVariableDef] = {}
     for kk, vv in def_outp_data.items():
         if vv.differentiable:
+            assert vv.category & OutputVariableOperation.DERV_R.value == 0
+            assert vv.category & OutputVariableOperation.DERV_C.value == 0
```

Comment:
we may take twice derivative over r in the future to compute the hessian. we may want to consider such extension.